### PR TITLE
Miha/Kevin Audio Download/Delete 

### PR DESCRIPTION
**Pull request by miha-bhaskaran**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

## :tada: What's new in this PR :tada:
Audio can now be downloaded to Expo File System and deleted from the File System.

### :cloud: :cloud: Description :cloud: :cloud:
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
With the audio ID, audio will be downloaded from Firebase into the Expo File System. We have also implemented a delete function that will delete the audio from the File System using the deleteAsync function that takes in the Audio's ID that needs to be deleted. 

### Screenshots :calling:
[//]: # "REQUIRED for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! If you are making changes here, please CC: @leexesther at the bottom."

## :zap: How to review :zap:
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
We tested this by integrating Audio Demo into the play screen. There, you can input any valid audio ID from cs-firebase and on terminal you should get log info messages that let you know if the audio exists in FileSystema and if its deleted. We set it up so that after an ID is inputted, it will get downloaded and then immediately deleted. These should be console logged to terminal to show if it exists or not.

Screenshot 2023-04-17 at 4 51 54 PM

## :seedling: Next steps :seedling:
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

## Relevant Links :link:
Expo File System Documentation: https://docs.expo.dev/versions/latest/sdk/filesystem/#saf-uri

### Online sources :information_source:
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs :raised_hands:
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @davidqing6432

### CODE REVIEW

File: src/firebase/AudioDemos.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+export default function QueriesDemo() {
+  const [audio] = useState<Audio[]>([]);
+  // eslint-disable-next-line camelcase
+  const [audio_id, setAudioId] = useState<string>('');
+  /* fetch all audio on page load */
+  const handlePress = async () => {
+    const audios = await downLoadAudio(audio_id);
+    const didItDownload = await getInfoAsync(audios);
+    // eslint-disable-next-line no-console
+    console.log(didItDownload);
```
Comment (david-qing) [inline src/firebase/AudioDemos.tsx]: isDownloaded instead

File: src/firebase/AudioDemos.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    deleteAudio(audios);
+    const didItDelete = await getInfoAsync(audios);
```
Comment (david-qing) [inline src/firebase/AudioDemos.tsx]: isDeleted instead

File: src/firebase/AudioDemos.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import { useState } from 'react';
+import { Button, Text, TextInput, View } from 'react-native';
+import { getInfoAsync } from 'expo-file-system';
+import { Audio } from '../types/schema';
+// import { getAllAudio, getAudioID } from './queries/audioQueries';
+import { downLoadAudio, deleteAudio } from './queries/audioPlayback';
+
+export default function QueriesDemo() {
+  const [audio] = useState<Audio[]>([]);
+  // eslint-disable-next-line camelcase
```
Comment (david-qing) [inline src/firebase/AudioDemos.tsx]: change to camelcase

File: src/firebase/AudioDemos.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
+/**
+ * Simple demo component that renders a list of all grants.
+ */
+
+import { useState } from 'react';
+import { Button, Text, TextInput, View } from 'react-native';
+import { getInfoAsync } from 'expo-file-system';
+import { Audio } from '../types/schema';
+// import { getAllAudio, getAudioID } from './queries/audioQueries';
```
Comment (david-qing) [inline src/firebase/AudioDemos.tsx]: remove

File: src/firebase/AudioDemos.tsx
Comment (david-qing) [inline src/firebase/AudioDemos.tsx:1]: we shouldn't be using an audio demos file, move this elsewhere or delete the file

File: src/firebase/queries/audioPlayback.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import { getAudioID } from './audioQueries';
+
+export const downLoadAudio = async (audioID: string): Promise<string> => {
+  const name = audioID;
+  const url = (await getAudioID(audioID)).gcsLink;
+  const dir = `${documentDirectory}${name}.mp3`;
+  const downLoadable = createDownloadResumable(url, dir, {});
+  try {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const uri = await downLoadable.downloadAsync();
```
Comment (david-qing) [inline src/firebase/queries/audioPlayback.ts]: what do we do with this uri?

File: src/firebase/queries/audioQueries.ts [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+ * Get all grants from the `grants` collection.
+ */
+export const getAllAudio = async (): Promise<Audio[]> => {
+  try {
+    const itemsRef = collection(db, 'audio');
+    const promises: Promise<Audio>[] = [];
+    const docSnap = await getDocs(itemsRef);
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    docSnap.forEach((audio: any) => {
```
Comment (david-qing) [inline src/firebase/queries/audioQueries.ts]: the audio should have a type

File: src/screens/PlayScreen/Play.tsx [hunk 7]
```diff
@@ -1,9 +1,46 @@
...
+} from '../../firebase/queries/audioPlayback';
+
+// export const getImage = async (audioID: string): Promise<string> => {
+//    const thumNail =  (await getAudioID(audioID)).thumbnail;
+//       return thumNail
+//  }
 
 function PlayScreen() {
+  const [audioId, setAudioId] = useState<string>('');
+  // eslint-disable-next-line no-var
```
Comment (david-qing) [inline src/screens/PlayScreen/Play.tsx]: remove

File: src/screens/PlayScreen/Play.tsx [hunk 8]
```diff
@@ -1,9 +1,46 @@
...
 import Colors from '../../styles/Colors';
 import styles from './styles';
+import { getAudioID } from '../../firebase/queries/audioQueries';
+
+import {
+  downLoadAudio,
+  deleteAudio,
+} from '../../firebase/queries/audioPlayback';
+
+// export const getImage = async (audioID: string): Promise<string> => {
```
Comment (david-qing) [inline src/screens/PlayScreen/Play.tsx]: remove

File: src/screens/PlayScreen/Play.tsx [hunk 9]
```diff
@@ -1,9 +1,46 @@
-import { Image, Text, View } from 'react-native';
+// import { Image, Text, View } from 'react-native';
```
Comment (david-qing) [inline src/screens/PlayScreen/Play.tsx]: remove

File: src/screens/PlayScreen/Play.tsx [hunk 10]
```diff
@@ -22,7 +59,19 @@ function PlayScreen() {
         </Text>
       </View>
       <View style={styles.container2}>
+        {/* <Image
```
Comment (david-qing) [inline src/screens/PlayScreen/Play.tsx]: remove

File: src/screens/PlayScreen/Play.tsx [hunk 11]
```diff
@@ -46,6 +95,15 @@ function PlayScreen() {
         <Icon type="play_button" />
         <Icon type="audio_forward" />
       </View>
+      <View
+        style={{
+          paddingBottom: '7%',
```
Comment (david-qing) [inline src/screens/PlayScreen/Play.tsx]: styles should be extracted to a file

File: src/screens/PlayScreen/Play.tsx [hunk 12]
```diff
@@ -1,9 +1,46 @@
...
+
+  //   a = holder
+  //   console.log('INSIDE');
+  //   console.log(a);
+  // });
+
+  /* fetch all audio on page load */
+  const handlePress = async () => {
+    const audios = await downLoadAudio(audioId);
+    const didItDownload = await getInfoAsync(audios);
```
Comment (david-qing) [inline src/screens/PlayScreen/Play.tsx]: isDownloaded

File: src/screens/PlayScreen/Play.tsx [hunk 13]
```diff
@@ -1,9 +1,46 @@
...
+    // eslint-disable-next-line no-console
+    console.log(didItDownload);
+
+    deleteAudio(audios);
+    const didItDelete = await getInfoAsync(audios);
```
Comment (david-qing) [inline src/screens/PlayScreen/Play.tsx]: isDeleted

File: yarn.lock
Comment (david-qing) [inline yarn.lock:1]: we should not have yarn; only use npm

File: App.tsx [hunk 14]
```diff
@@ -3,7 +3,10 @@ import * as Font from 'expo-font';
 import * as SplashScreen from 'expo-splash-screen';
 import { useCallback, useEffect, useState } from 'react';
 import { StyleSheet, View } from 'react-native';
+
 import RootNavigation from './src/navigation/RootNavigator';
+// import QueriesDemo from './src/firebase/QueriesDemo';
```
Comment (david-qing) [inline App.tsx]: these should be removed

File: App.tsx [hunk 15]
```diff
@@ -50,12 +53,14 @@ export default function App() {
...
       // we hide the splash screen once we know the root view has already
       // performed layout.
+
       await SplashScreen.hideAsync();
     }
   }, [resourcesLoaded]);
 
   return !resourcesLoaded ? null : (
     <View style={styles.container} onLayout={onLayoutRootView}>
+      {/* <AudioDemos /> */}
```
Comment (david-qing) [inline App.tsx]: remove

File: src/firebase/AudioDemos.tsx [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+    // eslint-disable-next-line no-console
+    console.log(didItDownload);
+
+    deleteAudio(audios);
+    const didItDelete = await getInfoAsync(audios);
+    // eslint-disable-next-line no-console
+    console.log(didItDelete);
+  };
+
+  // useEffect(() => {
```
Comment (david-qing) [inline src/firebase/AudioDemos.tsx]: should be removed

**Review by david-qing — APPROVED** [review]
looks good to merge!
